Replication of Ruyi's code.

Dataset: IXI

10% for test, 90% for training. (Option: k-folds cross validation, not implemented yet.)

Using 3D-CNN model code of my own

In [ ]:
# main:
!python preprocess.py

In [1]:
!python demo_1_1.py --num_epochs=500
!python demo_1_2.py --num_epochs=500 --MIN_TEST_LOSS=1000.0

l1_conv3d/Relu   [None, 130, 130, 110, 16]
l1_conv3d/max_pool3d   [None, 65, 65, 55, 16]
l2_conv3d/Relu   [None, 65, 65, 55, 32]
l2_conv3d/max_pool3d   [None, 33, 33, 28, 32]
l3_conv3d/Relu   [None, 33, 33, 28, 64]
l3_conv3d/max_pool3d   [None, 17, 17, 14, 64]
l4_conv3d/Relu   [None, 17, 17, 14, 64]
l4_conv3d/max_pool3d   [None, 9, 9, 7, 64]
l5_conv3d/Relu   [None, 9, 9, 7, 64]
l5_conv3d/max_pool3d   [None, 5, 5, 4, 64]
l6_fc/fc_out1   [None, 512]
l6_fc/dropout1/mul   [None, 512]
l7_fc/fc_out2   [None, 128]
l7_fc/dropout2/mul   [None, 128]
l8_fc/final_output   [None, 1]
2019-03-07 11:10:53.139948: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-03-07 11:10:53.235688: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-03-07 11:10:53.236056: I

model saved successfully.
Step 6700: training_loss = 384.69 (0.760 sec)
 (val_loss)test_loss = 317.10
Step 6800: training_loss = 232.61 (0.751 sec)
 (val_loss)test_loss = 318.35
Step 6900: training_loss = 389.88 (0.748 sec)
 (val_loss)test_loss = 311.66
Step 7000: training_loss = 378.10 (0.754 sec)
 (val_loss)test_loss = 313.34
Step 7100: training_loss = 393.65 (0.751 sec)
 (val_loss)test_loss = 320.28
Step 7200: training_loss = 299.04 (0.761 sec)
 (val_loss)test_loss = 310.40
Step 7300: training_loss = 326.95 (0.751 sec)
 (val_loss)test_loss = 309.50
best shot model: test_loss = 309.50, step = 7300
model saved successfully.
Step 7400: training_loss = 244.10 (0.749 sec)
 (val_loss)test_loss = 321.61
Step 7500: training_loss = 270.50 (0.750 sec)
 (val_loss)test_loss = 322.22
Step 7600: training_loss = 337.30 (0.748 sec)
 (val_loss)test_loss = 318.14
Step 7700: training_loss = 355.70 (0.748 sec)
 (val_loss)test_loss = 313.47
Step 7800: training_loss = 296.41 (0.750 sec)
 (val_loss)test_l

Step 16700: training_loss = 128.63 (0.801 sec)
 (val_loss)test_loss = 307.90
Step 16800: training_loss = 362.36 (0.758 sec)
 (val_loss)test_loss = 310.47
Step 16900: training_loss = 285.30 (0.748 sec)
 (val_loss)test_loss = 304.31
Step 17000: training_loss = 201.29 (0.776 sec)
 (val_loss)test_loss = 302.82
Step 17100: training_loss = 150.34 (0.780 sec)
 (val_loss)test_loss = 309.55
Step 17200: training_loss = 278.48 (0.753 sec)
 (val_loss)test_loss = 308.83
Step 17300: training_loss = 369.24 (0.742 sec)
 (val_loss)test_loss = 308.60
Step 17400: training_loss = 197.58 (0.746 sec)
 (val_loss)test_loss = 305.74
Step 17500: training_loss = 397.45 (0.749 sec)
 (val_loss)test_loss = 311.34
Step 17600: training_loss = 334.16 (0.753 sec)
 (val_loss)test_loss = 305.62
Step 17700: training_loss = 392.79 (0.747 sec)
 (val_loss)test_loss = 304.37
Step 17800: training_loss = 291.27 (0.747 sec)
 (val_loss)test_loss = 306.78
Step 17900: training_loss = 294.93 (0.757 sec)
 (val_loss)test_loss = 305.48

2019-03-07 16:50:25.935611: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2019-03-07 16:50:25.935639: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-07 16:50:25.935644: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2019-03-07 16:50:25.935648: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2019-03-07 16:50:25.935735: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10003 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Model loaded successfully.
test_loss = 300.26, test_accuracy = 0.00.
Figure(1500x500)
Figure(640x480)
MaxPool3D   [None, 65, 65, 55, 8]
MaxPool3D_1   [None, 33, 33, 28, 16]
MaxPool3D_2   [None, 17, 17, 14, 32]
MaxPool3D_3   [None, 9, 9, 7, 64]
MaxPool3D_4   [None, 5, 

Step 7300: training_loss = 1531.95 (0.475 sec)
 (val_loss)test_loss = 1138.93
Step 7400: training_loss = 1164.79 (0.466 sec)
 (val_loss)test_loss = 1117.62
Step 7500: training_loss = 1728.49 (0.465 sec)
 (val_loss)test_loss = 1255.55
Step 7600: training_loss = 1180.57 (0.473 sec)
 (val_loss)test_loss = 1153.23
Step 7700: training_loss = 1050.19 (0.464 sec)
 (val_loss)test_loss = 1129.30
Step 7800: training_loss = 1169.77 (0.464 sec)
 (val_loss)test_loss = 1107.94
Step 7900: training_loss = 1062.55 (0.475 sec)
 (val_loss)test_loss = 1077.67
Step 8000: training_loss = 1135.98 (0.468 sec)
 (val_loss)test_loss = 1068.49
Step 8100: training_loss = 1297.50 (0.465 sec)
 (val_loss)test_loss = 1153.50
Step 8200: training_loss = 1584.50 (0.475 sec)
 (val_loss)test_loss = 1124.58
Step 8300: training_loss = 907.43 (0.475 sec)
 (val_loss)test_loss = 1120.63
Step 8400: training_loss = 536.24 (0.465 sec)
 (val_loss)test_loss = 1139.87
Step 8500: training_loss = 1217.72 (0.464 sec)
 (val_loss)test_los

Step 17400: training_loss = 637.00 (0.463 sec)
 (val_loss)test_loss = 1009.45
Step 17500: training_loss = 908.07 (0.468 sec)
 (val_loss)test_loss = 946.06
Step 17600: training_loss = 1304.70 (0.465 sec)
 (val_loss)test_loss = 998.29
Step 17700: training_loss = 1212.00 (0.461 sec)
 (val_loss)test_loss = 945.93
Step 17800: training_loss = 897.02 (0.476 sec)
 (val_loss)test_loss = 848.95
best shot model: test_loss = 848.95, step = 17800
model saved successfully.
Step 17900: training_loss = 1369.28 (0.478 sec)
 (val_loss)test_loss = 972.68
Step 18000: training_loss = 997.25 (0.466 sec)
 (val_loss)test_loss = 929.94
Step 18100: training_loss = 1086.76 (0.469 sec)
 (val_loss)test_loss = 911.86
Step 18200: training_loss = 1102.25 (0.466 sec)
 (val_loss)test_loss = 1012.74
Step 18300: training_loss = 881.24 (0.464 sec)
 (val_loss)test_loss = 989.33
Step 18400: training_loss = 864.94 (0.463 sec)
 (val_loss)test_loss = 929.37
Step 18500: training_loss = 925.54 (0.466 sec)
 (val_loss)test_loss = 

l1_conv3d/Relu   [None, 130, 130, 110, 16]
l1_conv3d/max_pool3d   [None, 65, 65, 55, 16]
l2_conv3d/Relu   [None, 65, 65, 55, 32]
l2_conv3d/max_pool3d   [None, 33, 33, 28, 32]
l3_conv3d/Relu   [None, 33, 33, 28, 64]
l3_conv3d/max_pool3d   [None, 17, 17, 14, 64]
l4_conv3d/Relu   [None, 17, 17, 14, 64]
l4_conv3d/max_pool3d   [None, 9, 9, 7, 64]
l5_conv3d/Relu   [None, 9, 9, 7, 64]
l5_conv3d/max_pool3d   [None, 5, 5, 4, 64]
l6_fc/fc_out1   [None, 512]
l6_fc/dropout1/mul   [None, 512]
l7_fc/fc_out2   [None, 128]
l7_fc/dropout2/mul   [None, 128]
l8_fc/final_output   [None, 1]
2019-03-05 21:20:23.338765: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-03-05 21:20:23.422099: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-03-05 21:20:23.422458: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1411] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.721
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 10.29GiB
2019-03-05 21:20:23.422472: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2019-03-05 21:20:23.598899: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-05 21:20:23.598923: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2019-03-05 21:20:23.598929: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2019-03-05 21:20:23.599084: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9945 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Let's get started to train!
Step 0: training_loss = 2842651.50 (3.248 sec)
 (val_loss)test_loss = 1206660.00
Step 100: training_loss = 2136.75 (0.808 sec)
 (val_loss)test_loss = 2355.77
Step 200: training_loss = 3021.07 (0.771 sec)
 (val_loss)test_loss = 2332.73
Step 300: training_loss = 2711.74 (0.765 sec)
 (val_loss)test_loss = 2261.16
Step 400: training_loss = 3187.76 (0.753 sec)
 (val_loss)test_loss = 2226.69
Step 500: training_loss = 2148.18 (0.749 sec)
 (val_loss)test_loss = 2131.54
Step 600: training_loss = 2428.75 (0.758 sec)
 (val_loss)test_loss = 2050.34
Step 700: training_loss = 2519.15 (0.758 sec)
 (val_loss)test_loss = 1991.47
Step 800: training_loss = 1820.44 (0.758 sec)
 (val_loss)test_loss = 1922.60
Step 900: training_loss = 2895.77 (0.760 sec)
 (val_loss)test_loss = 1835.42
Step 1000: training_loss = 1933.29 (0.764 sec)
 (val_loss)test_loss = 1780.74
Step 1100: training_loss = 2272.71 (0.780 sec)
 (val_loss)test_loss = 1710.93
Step 1200: training_loss = 1004.35 (0.760 sec)
 (val_loss)test_loss = 1688.17
Step 1300: training_loss = 1575.36 (0.759 sec)
 (val_loss)test_loss = 1651.33
Step 1400: training_loss = 1685.21 (0.759 sec)
 (val_loss)test_loss = 1530.07
Step 1500: training_loss = 903.01 (0.752 sec)
 (val_loss)test_loss = 1449.46
Step 1600: training_loss = 1336.87 (0.754 sec)
 (val_loss)test_loss = 1434.45
Step 1700: training_loss = 1696.58 (0.757 sec)
 (val_loss)test_loss = 1363.41
Step 1800: training_loss = 1563.32 (0.754 sec)
 (val_loss)test_loss = 1339.14
Step 1900: training_loss = 1267.16 (0.762 sec)
 (val_loss)test_loss = 1311.07
Step 2000: training_loss = 1612.27 (0.764 sec)
 (val_loss)test_loss = 1212.67
Step 2100: training_loss = 1123.74 (0.756 sec)
 (val_loss)test_loss = 1115.69
Step 2200: training_loss = 1421.47 (0.761 sec)
 (val_loss)test_loss = 1098.95
Step 2300: training_loss = 999.21 (0.751 sec)
 (val_loss)test_loss = 1034.09
Step 2400: training_loss = 809.34 (0.759 sec)
 (val_loss)test_loss = 963.64
Step 2500: training_loss = 845.69 (0.765 sec)
 (val_loss)test_loss = 931.47
Step 2600: training_loss = 889.21 (0.764 sec)
 (val_loss)test_loss = 935.86
Step 2700: training_loss = 855.39 (0.758 sec)
 (val_loss)test_loss = 887.25
Step 2800: training_loss = 977.03 (0.768 sec)
 (val_loss)test_loss = 871.55
Step 2900: training_loss = 1026.75 (0.748 sec)
 (val_loss)test_loss = 785.97
Step 3000: training_loss = 513.43 (0.771 sec)
 (val_loss)test_loss = 695.72
Step 3100: training_loss = 258.84 (0.750 sec)
 (val_loss)test_loss = 716.96
Step 3200: training_loss = 581.67 (0.760 sec)
 (val_loss)test_loss = 643.52
Step 3300: training_loss = 667.14 (0.758 sec)
 (val_loss)test_loss = 597.73
Step 3400: training_loss = 901.19 (0.760 sec)
 (val_loss)test_loss = 603.12
Step 3500: training_loss = 718.87 (0.759 sec)
 (val_loss)test_loss = 599.23
Step 3600: training_loss = 521.17 (0.755 sec)
 (val_loss)test_loss = 538.84
Step 3700: training_loss = 676.75 (0.758 sec)
 (val_loss)test_loss = 520.04
Step 3800: training_loss = 467.73 (0.764 sec)
 (val_loss)test_loss = 454.41
best shot model: test_loss = 454.41, step = 3800
model saved successfully.
Step 3900: training_loss = 400.74 (0.762 sec)
 (val_loss)test_loss = 471.24
Step 4000: training_loss = 719.11 (0.762 sec)
 (val_loss)test_loss = 433.66
best shot model: test_loss = 433.66, step = 4000
model saved successfully.
Step 4100: training_loss = 464.01 (0.755 sec)
 (val_loss)test_loss = 430.95
best shot model: test_loss = 430.95, step = 4100
model saved successfully.
Step 4200: training_loss = 406.96 (0.768 sec)
 (val_loss)test_loss = 396.13
best shot model: test_loss = 396.13, step = 4200
model saved successfully.
Step 4300: training_loss = 404.16 (0.758 sec)
 (val_loss)test_loss = 419.16
Step 4400: training_loss = 271.93 (0.758 sec)
 (val_loss)test_loss = 408.86
Step 4500: training_loss = 349.34 (0.755 sec)
 (val_loss)test_loss = 401.01
Step 4600: training_loss = 354.61 (0.767 sec)
 (val_loss)test_loss = 403.39
Step 4700: training_loss = 316.45 (0.751 sec)
 (val_loss)test_loss = 371.88
best shot model: test_loss = 371.88, step = 4700
model saved successfully.
Step 4800: training_loss = 250.96 (0.757 sec)
 (val_loss)test_loss = 367.76
best shot model: test_loss = 367.76, step = 4800
model saved successfully.
Step 4900: training_loss = 485.94 (0.753 sec)
 (val_loss)test_loss = 373.31
Step 5000: training_loss = 666.73 (0.756 sec)
 (val_loss)test_loss = 373.46
Step 5100: training_loss = 197.77 (0.752 sec)
 (val_loss)test_loss = 333.66
best shot model: test_loss = 333.66, step = 5100
model saved successfully.
Step 5200: training_loss = 383.57 (0.765 sec)
 (val_loss)test_loss = 330.13
best shot model: test_loss = 330.13, step = 5200
model saved successfully.
Done training for 100 epochs, 5300 steps.
l1_conv3d/Relu   [None, 130, 130, 110, 16]
l1_conv3d/max_pool3d   [None, 65, 65, 55, 16]
l2_conv3d/Relu   [None, 65, 65, 55, 32]
l2_conv3d/max_pool3d   [None, 33, 33, 28, 32]
l3_conv3d/Relu   [None, 33, 33, 28, 64]
l3_conv3d/max_pool3d   [None, 17, 17, 14, 64]
l4_conv3d/Relu   [None, 17, 17, 14, 64]
l4_conv3d/max_pool3d   [None, 9, 9, 7, 64]
l5_conv3d/Relu   [None, 9, 9, 7, 64]
l5_conv3d/max_pool3d   [None, 5, 5, 4, 64]
l6_fc/fc_out1   [None, 512]
l6_fc/dropout1/mul   [None, 512]
l7_fc/fc_out2   [None, 128]
l7_fc/dropout2/mul   [None, 128]
l8_fc/final_output   [None, 1]
2019-03-05 22:28:45.091585: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2019-03-05 22:28:45.091617: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-05 22:28:45.091622: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2019-03-05 22:28:45.091626: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2019-03-05 22:28:45.091709: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9945 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Model loaded successfully.
test_loss = 330.13, test_accuracy = 0.00.
MaxPool3D   [None, 65, 65, 55, 8]
MaxPool3D_1   [None, 33, 33, 28, 16]
MaxPool3D_2   [None, 17, 17, 14, 32]
MaxPool3D_3   [None, 9, 9, 7, 64]
MaxPool3D_4   [None, 5, 5, 4, 128]
dropout_5/mul   [None, 1024]
output   [None, 1]

2019-03-05 22:28:50.843279: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-03-05 22:28:50.919529: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-03-05 22:28:50.919899: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1411] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.721
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 10.23GiB
2019-03-05 22:28:50.919913: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2019-03-05 22:28:51.097448: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-05 22:28:51.097474: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2019-03-05 22:28:51.097479: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2019-03-05 22:28:51.097635: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9886 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Let's get started to train!
Step 0: training_loss = 7494902.50 (2.231 sec)
 (val_loss)test_loss = 2981516.25
Step 100: training_loss = 731.93 (0.475 sec)
 (val_loss)test_loss = 6703.18
Step 200: training_loss = 63865.95 (0.483 sec)
 (val_loss)test_loss = 53182.90
Step 300: training_loss = 2457.39 (0.473 sec)
 (val_loss)test_loss = 6459.66
Step 400: training_loss = 33791.54 (0.466 sec)
 (val_loss)test_loss = 8740.79
Step 500: training_loss = 1886.30 (0.473 sec)
 (val_loss)test_loss = 2564.95
Step 600: training_loss = 2656.52 (0.476 sec)
 (val_loss)test_loss = 3104.27
Step 700: training_loss = 2253.85 (0.471 sec)
 (val_loss)test_loss = 1894.96
Step 800: training_loss = 1809.35 (0.464 sec)
 (val_loss)test_loss = 4132.34
Step 900: training_loss = 2170.93 (0.472 sec)
 (val_loss)test_loss = 1752.95
Step 1000: training_loss = 1740.30 (0.475 sec)
 (val_loss)test_loss = 1774.45
Step 1100: training_loss = 2301.06 (0.485 sec)
 (val_loss)test_loss = 2691.19
Step 1200: training_loss = 3045.00 (0.470 sec)
 (val_loss)test_loss = 3899.98
Step 1300: training_loss = 1093.60 (0.467 sec)
 (val_loss)test_loss = 1379.85
Step 1400: training_loss = 3520.94 (0.466 sec)
 (val_loss)test_loss = 1519.44
Step 1500: training_loss = 816.80 (0.481 sec)
 (val_loss)test_loss = 2046.40
Step 1600: training_loss = 1087.52 (0.481 sec)
 (val_loss)test_loss = 1882.73
Step 1700: training_loss = 1620.36 (0.470 sec)
 (val_loss)test_loss = 1622.95
Step 1800: training_loss = 2187.07 (0.477 sec)
 (val_loss)test_loss = 1927.22
Step 1900: training_loss = 988.63 (0.472 sec)
 (val_loss)test_loss = 1552.87
Step 2000: training_loss = 2057.38 (0.474 sec)
 (val_loss)test_loss = 1612.64
Step 2100: training_loss = 1195.94 (0.481 sec)
 (val_loss)test_loss = 1908.41
Step 2200: training_loss = 1796.98 (0.468 sec)
 (val_loss)test_loss = 1394.80
Step 2300: training_loss = 1173.94 (0.469 sec)
 (val_loss)test_loss = 1789.19
Step 2400: training_loss = 1003.54 (0.485 sec)
 (val_loss)test_loss = 1456.03
Step 2500: training_loss = 664.47 (0.475 sec)
 (val_loss)test_loss = 788.31
Step 2600: training_loss = 897.41 (0.470 sec)
 (val_loss)test_loss = 973.23
Step 2700: training_loss = 1465.58 (0.468 sec)
 (val_loss)test_loss = 2184.36
Step 2800: training_loss = 1122.05 (0.476 sec)
 (val_loss)test_loss = 914.04
Step 2900: training_loss = 1726.58 (0.469 sec)
 (val_loss)test_loss = 1305.81
Step 3000: training_loss = 1117.64 (0.469 sec)
 (val_loss)test_loss = 1459.36
Step 3100: training_loss = 443.60 (0.462 sec)
 (val_loss)test_loss = 1284.50
Step 3200: training_loss = 1437.12 (0.465 sec)
 (val_loss)test_loss = 1386.12
Step 3300: training_loss = 1011.89 (0.467 sec)
 (val_loss)test_loss = 1281.37
Step 3400: training_loss = 1349.02 (0.486 sec)
 (val_loss)test_loss = 1226.90
Step 3500: training_loss = 1162.26 (0.472 sec)
 (val_loss)test_loss = 1353.63
Step 3600: training_loss = 1312.94 (0.468 sec)
 (val_loss)test_loss = 1714.27
Step 3700: training_loss = 1738.36 (0.467 sec)
 (val_loss)test_loss = 1375.46
Step 3800: training_loss = 1287.00 (0.469 sec)
 (val_loss)test_loss = 1272.01
Step 3900: training_loss = 1276.78 (0.468 sec)
 (val_loss)test_loss = 1352.48
Step 4000: training_loss = 1779.12 (0.475 sec)
 (val_loss)test_loss = 1343.68
Step 4100: training_loss = 1848.59 (0.467 sec)
 (val_loss)test_loss = 1688.99
Step 4200: training_loss = 857.53 (0.469 sec)
 (val_loss)test_loss = 1022.35
Step 4300: training_loss = 1294.62 (0.478 sec)
 (val_loss)test_loss = 1219.50
Step 4400: training_loss = 882.61 (0.482 sec)
 (val_loss)test_loss = 1180.66
Step 4500: training_loss = 1033.62 (0.483 sec)
 (val_loss)test_loss = 1352.14
Step 4600: training_loss = 1358.41 (0.468 sec)
 (val_loss)test_loss = 1661.28
Step 4700: training_loss = 1317.87 (0.467 sec)
 (val_loss)test_loss = 1323.33
Step 4800: training_loss = 812.51 (0.468 sec)
 (val_loss)test_loss = 1150.15
Step 4900: training_loss = 985.01 (0.477 sec)
 (val_loss)test_loss = 1159.69
Step 5000: training_loss = 1890.36 (0.471 sec)
 (val_loss)test_loss = 1231.50
Step 5100: training_loss = 683.88 (0.468 sec)
 (val_loss)test_loss = 1138.48
Step 5200: training_loss = 1190.55 (0.471 sec)
 (val_loss)test_loss = 1091.40
Done training for 100 epochs, 5300 steps.
MaxPool3D   [None, 65, 65, 55, 8]
MaxPool3D_1   [None, 33, 33, 28, 16]
MaxPool3D_2   [None, 17, 17, 14, 32]
MaxPool3D_3   [None, 9, 9, 7, 64]
MaxPool3D_4   [None, 5, 5, 4, 128]
dropout_5/mul   [None, 1024]
output   [None, 1]
2019-03-05 23:11:34.694014: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2019-03-05 23:11:34.694046: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-05 23:11:34.694051: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2019-03-05 23:11:34.694055: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2019-03-05 23:11:34.694138: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9886 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
Traceback (most recent call last):
  File "demo_1_2.py", line 358, in <module>
    tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
  File "/home/woody/vpython/vp3.5.2/lib/python3.5/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "demo_1_2.py", line 303, in main
    run_test()
  File "demo_1_2.py", line 265, in run_test
    saver.restore(sess, FLAGS.saver_dir)
  File "/home/woody/vpython/vp3.5.2/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 1530, in restore
    + compat.as_text(save_path))
ValueError: The passed save_path is not a valid checkpoint: ./log/demo1.2_ruyi_model.ckpt


In [ ]:
%load_ext autoreload
%autoreload 2



In [ ]:
from my_tools import *
plt_data_path_name = './img/demo_1_1_pltdata_' + time_now()
if not os.path.exists(plt_data_path_name + '.npy'):
    np.save(plt_data_path_name, np.array([[1,2],[3,4]]))
else:
    print(plt_data_path_name + '.npy exists already.')

In [ ]:
import numpy as np
t = time.strftime?
type(t)
a = np.array([[1,2],[3,4]])
np.save('./test'+t,a)
import os
print(np.log10([1000,2,3]))
print(np.log([1000,2,3]))

In [ ]:
npy_img = np.load('./IXI_npy/origin/IXI012.npy')
print(npy_img.shape)
plt.imshow(npy_img[:,:,60])
# plt.imshow(npy_img[:,60,:])
# plt.imshow(npy_img[60,:,:])

In [ ]:
a = np.random.randint(10,size=(3,3))
b = np.random.randint(10,size=(3,3))
c = np.random.randn(3,3)
print(c)
# a += 1
# print(a)
cc = c.astype(int)
print(cc.dtype)
# print(c)
# print(c)
# c.astype(int)
# preprocess_1('IXI002-Guys-0828-T1.nii.gz')

In [ ]:
10/3

In [ ]:
a = np.random.randint

In [ ]:
phenotypics = pd.read_csv('./phenotypics.csv', sep=',',header=0)
# print(phenotypics)
phenotypics[5:10]

In [ ]:
!cd /media/woody/Elements/age_data/IXI/IXI-T1/

os.path.exists('/media/woody/Elements/age_data/IXI/IXI-T1/IXI.*-.*-.*-T1\.nii\.gz$')

In [ ]:
import time
from progressbar import *

# for i in range(10):
#     print('\r'+str((i+1)*100.0/10)+'%',end='',flush=True)
#     time.sleep(0.3)
# print('abc')

pbar = ProgressBar().start()
for i in range(10):
    pbar.update(int(i*100/(10-1)))
    time.sleep(0.8)
pbar.finish()

In [ ]:
import time
from progressbar import *
total = 100
def dosomework():
    time.sleep(0.01)
widgets = ['Progress: ',Percentage(), ' ', Bar('>'),' ', Timer(),
      ' ', ETA()]
pbar = ProgressBar(widgets=widgets, maxval=total+100).start()
for i in range(total):
    # do something
    pbar.update(i)
    dosomework()
pbar.finish()

In [ ]:
ProgressBar?

In [ ]:
  
import sys, time  
from progressbar import *  
  
total = 1000  
  
def dosomework():  
    time.sleep(0.01)  
pbar = ProgressBar().start()  
for i in range(1000):  
    pbar.update(int((i / (total - 1)) * 100))  
    dosomework()  
pbar.finish()  

In [ ]:
data = xlrd.open_workbook('IXI.xls','rb')
print('工作表名为：'+ data.sheet_names()[0])
table = data.sheets()[0]
nrows = table.nrows
ncols = table.ncols
print('表格行数为%d,列数为%d'%(nrows,ncols))

#输出每一行的值
# for item in range(table.nrows):
#     print(table.row_values(item))

#获取单元格的值
cell_A1 = table.row(0)[0].value
cell_A2 = table.cell(0,0).value
cell_A3 = table.col(0)[0].value

print(cell_A1)
print(cell_A2)
print(cell_A3)